Fix feedrate after toolchange

Blue-Marlin opened the pull request:
The fix is simple. 
Most of the changes are because of changed indentation,
bacause of leaving early a for wrong tool number.
https://github.com/MarlinFirmware/Marlin/pull/3620/files?w=1

Fix for #3610

Alex9779 commented:
Thanks, when I got time, maybe later today...

Alex9779 commented:
Ok I tested. This PR fixes the issue I reported in #3610...

thinkyhead commented on Marlin/Marlin_main.cpp:
```diff
@@ -6304,131 +6304,136 @@ inline void gcode_T(uint8_t tmp_extruder) {
-        make_move = true;
-      #endif
+  #if EXTRUDERS > 1
+    target_extruder = tmp_extruder;
```
It looks like this line doesn't do anything, so it can probably be dropped.

thinkyhead commented on Marlin/Marlin_main.cpp:
```diff
@@ -6304,131 +6304,136 @@ inline void gcode_T(uint8_t tmp_extruder) {
+        }
+        else if (dual_x_carriage_mode == DXC_DUPLICATION_MODE) {
+          active_extruder_parked = (active_extruder == 0); // this triggers the second extruder to move into the duplication position
+          if (active_extruder == 0 || active_extruder_parked)
```
Because `active_extruder_parked = (active_extruder == 0)`…

``` cpp
if (active_extruder_parked) ...
```

Blue-Marlin committed:
Fix feedrate after toolchange

The fix is simple. Most changes are because of changed indendation,
bacause of leaving early for wrong tool number

Blue-Marlin commented:
@thinkyhead 
Updated.

thinkyhead commented:
This is nice and comprehensive. I have my fingers crossed that it works well.
We don't get to test DUAL_X nearly enough!

thinkyhead commented:
> Most of the changes are because of changed indentation

I have found it helpful in such cases to do the code changes first, but leave indentation the same, commit that first. And then commit the indentation changes (elimination of `else`, adding `return`, etc.) afterward. Then at least the first commit makes a more readable diff.
